src/hotspot/share/runtime/javaCalls.cpp

```diff
@@ -398,20 +390,35 @@ void JavaCalls::call_helper(JavaValue* result, const methodHandle& method, JavaC
       // the call to call_stub, the optimizer produces wrong code.
       intptr_t* result_val_address = (intptr_t*)(result->get_value_addr());
       intptr_t* parameter_address = args->parameters();
+
+      address entry_point;
+      {
+        // The enter_interp_only_mode use handshake to set interp_only mode
+        // so no safepoint should be allowed between is_interp_only_mode() and call
+        NoSafepointVerifier nsv;
+        if (JvmtiExport::can_post_interpreter_events() && thread->is_interp_only_mode()) {
+          entry_point = method->interpreter_entry();
+        } else {
+          // Since the call stub sets up like the interpreter we call the from_interpreted_entry
+          // so we can go compiled via a i2c.
+          entry_point = method->from_interpreted_entry();
 #if INCLUDE_JVMCI
-      // Gets the alternative target (if any) that should be called
-      Handle alternative_target = args->alternative_target();
-      if (!alternative_target.is_null()) {
-        // Must extract verified entry point from HotSpotNmethod after VM to Java
-        // transition in JavaCallWrapper constructor so that it is safe with
-        // respect to nmethod sweeping.
-        address verified_entry_point = (address) HotSpotJVMCI::InstalledCode::entryPoint(nullptr, alternative_target());
-        if (verified_entry_point != nullptr) {
-          thread->set_jvmci_alternate_call_target(verified_entry_point);
-          entry_point = method->adapter()->get_i2c_entry();
+          // Gets the alternative target (if any) that should be called
+          Handle alternative_target = args->alternative_target();
+          if (!alternative_target.is_null()) {
+            // Must extract verified entry point from HotSpotNmethod after VM to Java
+            // transition in JavaCallWrapper constructor so that it is safe with
+            // respect to nmethod sweeping.
+            address
+                verified_entry_point = (address) HotSpotJVMCI::InstalledCode::entryPoint(nullptr, alternative_target());
```

Comment:
This line-break choice seems unconventional.  Can we just make it a single line like before?